File: Python/Test/jdi_uitests_webtests/main/page_objects/sections/jdi_search.py
from JDI.web.selenium.elements.api_interact.find_element_by import By
from JDI.web.selenium.elements.common.button import Button
from JDI.web.selenium.elements.common.text_field import TextField
from JDI.web.selenium.elements.composite.search import Search


class JDISearch(Search):
    search_input = TextField(By.css(".search-field input"))
    search_button = Button(By.css(".search .search-active .icon-search"))

    def find(self, text):
        Button(By.css(".search>.icon-search")).click()
        super(JDISearch, self).find(text)

File: Java/Tests/jdi-uitest-webtests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/sections/JdiSearch.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.sections;

import com.epam.jdi.uitests.core.interfaces.common.IButton;
import com.epam.jdi.uitests.core.interfaces.common.ITextField;
import com.epam.jdi.uitests.web.selenium.elements.common.Button;
import com.epam.jdi.uitests.web.selenium.elements.composite.Search;
import org.openqa.selenium.By;
import org.openqa.selenium.support.FindBy;

/**
 * Created by Roman_Iovlev on 12/17/2015.
 */
public class JdiSearch extends Search {
    @FindBy(css = ".search-field input")
    public ITextField searchInput;
    //It does not work for now
    @FindBy(xpath = "//span[@class='icon-search active']")
    public IButton searchButton;

    @Override
    public void findAction(String text) {
        new Button(By.cssSelector(".search>.icon-search")).click();
        searchInput.newInput(text);
        //super.findAction(text);
        searchButton.click();
    }

}
